src/com/facebook/buck/android/AndroidResource.java

```diff
@@ -367,8 +377,23 @@ public BuildableProperties getProperties() {
 
   @Override
   public Sha1HashCode getAbiKeyForDeps() {
+    ImmutableSet<HasAndroidResourceDeps> resourceDeps;
+    switch (resourceDependencyMode) {
+      case FIRST_ORDER:
+        resourceDeps = ImmutableSet.copyOf(getNonEmptyResourceDeps());
+        break;
+      case TRANSITIVE:
+        resourceDeps = UnsortedAndroidResourceDeps.createFrom(
+            getDeps(),
+            Optional.<UnsortedAndroidResourceDeps.Callback>absent())
+                       .getResourceDeps();
```

Comment:
@natthu spotted that we don't make sure these are available with top-down-building (the default).  I've got a fix locally, but @natthu wants to try to fix something in the graph enhancement that will make my fix less awful, I think.  We should be able to get this in real soon.